File: app/src/main/java/s15666/pjwstk/pamobmi/BmiCategory.java
package s15666.pjwstk.pamobmi;

enum BmiCategory {

    UNDERWEIGHT("Underweight", 18.5),
    NORMAL("Normal weight", 24.9),
    OVERWEIGHT("Overweight", 29.9),
    OBESE("Obese", Double.MAX_VALUE);

    BmiCategory(String name, double max) {
        this.name = name;
        this.max = max;
    }

    // TODO color
    private final String name;
    private final double max;

    public double getMax() {
        return max;
    }

    @Override
    public String toString() {
        return name;
    }

    static BmiCategory getCategory(double bmi) throws NoSuchBmiException {
        BmiCategory[] categories = BmiCategory.values();
        for (BmiCategory c : categories) {
            if (bmi < c.getMax()) {
                return c;
            }
        }
        throw new NoSuchBmiException();
    }

    static class NoSuchBmiException extends Exception {
    }

}

File: app/src/main/java/s15666/pjwstk/pamobmi/ui/bmi/BmiCategory.kt
package s15666.pjwstk.pamobmi.ui.bmi

enum class BmiCategory(private val description: String, val max: Double) {
    UNDERWEIGHT("Underweight", 18.5),
    NORMAL("Normal weight", 24.9),
    OVERWEIGHT("Overweight", 29.9),
    OBESE("Obese", Double.MAX_VALUE);

    override fun toString(): String {
        return description
    }

    internal class NoSuchBmiException : Exception()
    companion object {
        @Throws(NoSuchBmiException::class)
        fun getCategory(bmi: Double): BmiCategory {
            val categories = values()
            for (c in categories) {
                if (bmi < c.max) {
                    return c
                }
            }
            throw NoSuchBmiException()
        }
    }

}